src/main/kotlin/org/rust/ide/console/RsConsoleRunner.kt

```diff
@@ -0,0 +1,226 @@
+                RsConsoleToolWindow.getInstance(project).hide()
+            }
+        }
+        return needInstallEvcxr
+    }
+
+    override fun initAndRun() {
+        UIUtil.invokeAndWaitIfNeeded(Runnable {
+            super.initAndRun()
+
+            consoleExecuteActionHandler.sendText(":opt 0\n")
+        })
+    }
+
+    override fun createProcessHandler(process: Process): OSProcessHandler =
+        RsConsoleProcessHandler(
+            process,
+            consoleView,
+            consoleCommunication,
+            commandLine!!.commandLineString,
+            StandardCharsets.UTF_8
+        )
+
+    private fun createCommandLine(): GeneralCommandLine? {
+        val cargoProject = project.cargoProjects.allProjects.firstOrNull()
+        val workingDir = cargoProject?.workingDirectory ?: return null
+        return project.toolchain?.evcxr()?.createCommandLine(workingDir.toFile())
+    }
+
+    override fun createProcess(): Process = commandLine!!.createProcess()
+
+    private fun connect() {
+        ApplicationManager.getApplication().invokeLater {
+            consoleView.executeActionHandler = consoleExecuteActionHandler
+
+            consoleExecuteActionHandler.isEnabled = true
+
+            consoleView.initialized()
+        }
+    }
+
+    override fun createExecuteActionHandler(): RsConsoleExecuteActionHandler {
+        val consoleExecuteActionHandler =
+            RsConsoleExecuteActionHandler(processHandler!!, this, consoleCommunication, consoleView)
+        consoleExecuteActionHandler.isEnabled = false
+        return consoleExecuteActionHandler
+    }
+
+    fun rerun() {
+        object : Task.Backgroundable(project, "Restarting Console", true) {
+            override fun run(indicator: ProgressIndicator) {
+                val processHandler = processHandler
+                if (processHandler != null) {
+                    processHandler.destroyProcess()
+                    processHandler.waitFor()
+                }
+
+                GuiUtils.invokeLaterIfNeeded({
+                    val runner = RsConsoleRunnerFactory.getInstance().createConsoleRunner(project, null)
+                    runner.run(true)
+                }, ModalityState.defaultModalityState())
+            }
+        }.queue()
+    }
+
+    private fun showErrorsInConsole(e: Exception) {
+        val actionGroup = DefaultActionGroup(RestartAction(this))
+
+        val actionToolbar = ActionManager.getInstance()
+            .createActionToolbar("RsConsoleRunnerErrors", actionGroup, false)
+
+        // Runner creating
+        val panel = JPanel(BorderLayout())
+        panel.add(actionToolbar.component, BorderLayout.WEST)
+
+        val errorViewPanel = NewErrorTreeViewPanel(project, null, false, false, null)
+
+        val messages = mutableListOf("Can't start evcxr. Please install it using `cargo install evcxr`")
+        if (StringUtil.isNotEmpty(e.message)) {
+            messages += StringUtil.splitByLines(e.message!!)
```

Comment:
Please check the nullability here. The usual way to do it in Kotlin is extracting variable (instead of checking nullability a property with getter and using `!!`), like this:
```Kotlin
val message = e.message
if (message != null && message.isNotBlank()) { // you can use Kotlin's `isNotBlank` (or `isNotEmpty`, if you don't want to skip blank lines as well as empty ones) instead of `StringUtil`)
    messages += message.lines() // you can use Kotlin's `lines()` instead of `StringUtil`
}
```